cilium-cli/connectivity/builder/strict_mode_encryption.go

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package builder
+
+import (
+	"github.com/cilium/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium/cilium-cli/connectivity/tests"
+	"github.com/cilium/cilium/cilium-cli/utils/features"
+)
+
+type strictModeEncryption struct{}
+
+func (t strictModeEncryption) build(ct *check.ConnectivityTest, _ map[string]string) {
+	newTest("strict-mode-encryption", ct).
+		WithCondition(func() bool { return ct.Params().IncludeUnsafeTests }).
+		WithFeatureRequirements(
+			features.RequireEnabled(features.EncryptionStrictMode),
+			// Strict mode is only supported with WireGuard
+			features.RequireMode(features.EncryptionPod, "wireguard"),
+			// Strict mode always allows host-to-host tunnel traffic
+			features.RequireDisabled(features.Tunnel),
+		).
```

Comment:
Now that we rely on the `cilium-dbg` command to manipulate the IP cache, this would also need to limit to latest-greatest Cilium version.

Reply:
indeed, I'm just working and testing locally, thank you~